### PR TITLE
262 flags use dashes

### PR DESCRIPTION
**Pull request by brahmlower**

### CODE REVIEW

File: src/base-command.ts [hunk 1]
```diff
@@ -16,6 +20,15 @@ export default abstract class extends Command {
     help: flags.help({ char: 'h' }),
   };
 
+  checkFlagDeprecations(flags: any, flagDefinitions: any) {
+    Object.keys(flags).forEach((flagName: string) => {
+      const description: string = flagDefinitions[flagName].description
+      if (description?.startsWith(DEPRECATED_LABEL)) {
```
Comment (tjhiggins) [inline src/base-command.ts:26]: Fix lint warnings

File: src/base-command.ts [hunk 2]
```diff
@@ -16,6 +20,15 @@ export default abstract class extends Command {
     help: flags.help({ char: 'h' }),
   };
 
+  checkFlagDeprecations(flags: any, flagDefinitions: any) {
```
Comment (tjhiggins) [inline src/base-command.ts:23]: Something I didn't cover yet, but our coding style is:

snakecase for vars
camelcase for functions

File: src/base-command.ts [hunk 3]
```diff
@@ -16,6 +20,15 @@ export default abstract class extends Command {
...
+    Object.keys(flags).forEach((flagName: string) => {
+      const description: string = flagDefinitions[flagName].description
+      if (description?.startsWith(DEPRECATED_LABEL)) {
+        this.warn(`Flag --${flagName} is deprecated.${description.split(DEPRECATED_LABEL)[1]}`);
+      }
+    });
+  }
+
```
Comment (tjhiggins) [inline src/base-command.ts:31]: I see my pseudo code didn't work first time. Use the following so you don't have to call checkFlagDeprecations in every command.

```ts
async init() {
    const { flags } = this.parse(this.constructor as any);
    const flag_definitions = (this.constructor as any).flags;
    this.checkFlagDeprecations(flags, flag_definitions);
    ...
```

File: src/commands/platforms/create.ts [hunk 4]
```diff
@@ -24,24 +24,79 @@ export default class PlatformCreate extends Command {
...
+    }),
+    aws_region: flags.string({
+      exclusive: ['awsconfig', 'kubeconfig', 'service-token', 'cluster-ca-cert', 'host'],
+      description: `${Command.DEPRECATED} Please use --aws-region.`,
+      hidden: true,
+    }),
+    ['aws-region']: flags.string({
+      exclusive: ['awsconfig', 'kubeconfig', 'service-token', 'cluster-ca-cert', 'host']
+    }),
+    service_token: flags.string({
```
Comment (tjhiggins) [inline src/commands/platforms/create.ts:63]: Remove unused flags service_token/cluster_ca_cert and their new dashed versions

**Review by tjhiggins — CHANGES_REQUESTED** [review]

File: src/base-command.ts (hunk 1 above)
Comment (tjhiggins) [reply src/base-command.ts:26]: I spoke too soon - appreciate the self review :)

File: src/base-command.ts (hunk 3 above)
Comment (brahmlower) [reply src/base-command.ts:31]: doah, that helps a bunch, thanks! Will get that switched over

File: src/base-command.ts (hunk 2 above)
Comment (brahmlower) [reply src/base-command.ts]: oh thanks- I forgot to ask about this earlier. I did another quick review and I think this function had the only styling violations. Should be fixed in the latest commit.

File: src/commands/platforms/create.ts (hunk 4 above)
Comment (brahmlower) [reply src/commands/platforms/create.ts:63]: If it's okay with you, I'd really like to remove these in a separate commit, that way the resulting merge commits are relatively atomic, even though it kinda sucks to maintain the dead weight of these couple extra flags through this PR 😬

Comment (tjhiggins) [reply src/commands/platforms/create.ts:63]: Fine by me as long as we don't forget

Comment (brahmlower) [reply src/commands/platforms/create.ts:63]: Gotcha- I recognize I still haven't made a ticket for taking care of that. Meant to make it after wrapping up this PR, but this has taken a fair bit longer than expected 😅

**Pull request merged**